pinot-query-runtime/src/main/java/org/apache/pinot/query/mailbox/GrpcSendingMailbox.java

```diff
@@ -48,6 +49,11 @@ public void init()
     PinotMailboxGrpc.PinotMailboxStub stub = PinotMailboxGrpc.newStub(channel);
     _statusStreamObserver = new MailboxStatusStreamObserver();
     _statusStreamObserver.init(stub.open(_statusStreamObserver));
+    // send a begin-of-stream message.
+    _statusStreamObserver.send(MailboxContent.newBuilder()
```

Comment:
Can you elaborate on why this is needed ? As long as the GRPC error is translated / converted into the error block (METADATA block type in the last PR) and sent by the sender, it should work just like how you handled other failures in the last PR https://github.com/apache/pinot/pull/9064/ ?

Reply:
if the mailbox sender/receiver pair is planned but the first message contains exception. the mailbox will not be closed until timeout. this will result in stall mailbox objects and potentially unclosed grpc connections. 

so we send the first begin-of-stream message that guarantees no error will occur with that payload, thus establish the connection (it is like a handshake on the mailbox layer)

Reply:
Got it. Thanks